### ISSUE TITLE
Reset button to Serial Monitor

### ISSUE THREAD

**Issue by night-ghost** [arduino/Arduino#4353]
Sometimes hardware Reset button unavailable (Arduino via RDP for example) so needs ability to reset without reflash

**Comment by NicoHood** [comment]
What a simple feature. Never thought of this, but it makes totally sense. I like the idea.

**Comment by matthijskooijman** [comment]
This button should probably be called "Reconnect", though, since boards do not necessarily reset on (re)connect (most though, but not all).

**Comment by NicoHood** [comment]
Or we just add buttons for DTR and RTS. Then the user can manually pull those lines, for those who want to hold the line down. I heard that some people need this for ESP anyways. Then its even more functional.

**Comment by matthijskooijman** [comment]
That would work (See #3386), though it might be less novice-friendly perhaps.

**Comment by NicoHood** [comment]
That is totally different from what the user wants here.
We do not want to set DTR/RTS at the beginning fixed, we want to change it at runtime in the serial monitor, to actually reset the mcu.

**Comment by matthijskooijman** [comment]
@NicoHood, that issue also talks about having the pins GUI-configurable, though I now see it's not the original request. Should have read it better before linking :-)

**Comment by chrstphrchvz** [comment]
Another user here. One workaround (at least on OS X) is to use baud pulldown menu and select same rate again, which resets the connection, and resets the board if auto-reset is enabled on it.

**Comment by chrstphrchvz** [comment]
I think the idea of the "Reconnect" button is helpful in the case where the board was unplugged, then reattached. The serial monitor doesn't automatically reconnect in that case, nor does it acknowledge that the board was unplugged, but a "Reconnect" button is a convenience over reopening the serial monitor.

**Comment by lpdunwell** [comment]
great. so in nearly 3 years this has not been implemented?

**Comment by alexceltare2** [comment]
> great. so in nearly 3 years this has not been implemented?

I agree, the IDE has plenty of spare "square" buttons besides Verify, Upload, New, Open and Save. Also the Serial monitor can add an extra button for "Reset Target" as well.
